File: src/main/java/FlutterContext.java
import com.intellij.codeInsight.template.TemplateContextType;
import com.intellij.codeInsight.template.TemplateActionContext;
import com.intellij.codeInsight.template.TemplateContextType;
import org.jetbrains.annotations.NotNull;

public class FlutterContext extends TemplateContextType {
    protected FlutterContext() {
        super("FLUTTER_RIVERPOD_SNIPPETS", "Flutter riverpod snippets");
    }

    @Override
    public boolean isInContext(@NotNull TemplateActionContext templateActionContext) {
        return templateActionContext.getFile().getName().endsWith(".dart");
    }
}

File: src/main/kotlin/dev/tbm98/flutterriverpodsnippets/FlutterContext.kt
package dev.tbm98.flutterriverpodsnippets

import com.intellij.codeInsight.template.TemplateActionContext
import com.intellij.codeInsight.template.TemplateContextType

class FlutterContext private constructor() :
    TemplateContextType("FLUTTER_RIVERPOD_SNIPPETS", "Flutter riverpod snippets") {
    override fun isInContext(templateActionContext: TemplateActionContext): Boolean {
        return templateActionContext.file.name.endsWith(".dart")
    }
}
